### PR TITLE
[JENKINS-64124] Support detecting automatic build with causes

### PR DESCRIPTION
**Pull request by gmshake**
This is partial implementation to support @agabrys 's request. See https://github.com/jenkinsci/branch-api-plugin/pull/186 for full context.

Created a Jira issue to track this, https://issues.jenkins-ci.org/browse/JENKINS-64124

### CODE REVIEW

File: src/main/java/jenkins/branch/BranchBuildStrategy.java [hunk 1]
```diff
@@ -160,13 +161,45 @@ public boolean isAutomaticBuild(@NonNull SCMSource source,
...
+     *                     the case of non {@link SCMRevision#isDeterministic()} previous revisions as polling for changes will have
+     *                     confirmed that there is a change between this and {@code currRevision} even if the two are equal.
+     * @param lastSeenRevision the {@link SCMRevision} that the head was last seen
+     * @param listener     the {@link TaskListener} that can be used for outputting any rational for the decision
+     * @param causes       the {@link Cause} of the build
+     * @return {@code true} if and only if the {@link SCMRevision} should be automatically built when the
+     * {@link SCMHead} has been detected as created / modified.
+     * @since 2.5.7
+     */
+    @SuppressWarnings("deprecation")
```
Comment (bitwiseman) [inline src/main/java/jenkins/branch/BranchBuildStrategy.java:194]:
```suggestion
```

**Comment by gmshake** [comment]
Hi, @bitwiseman 
> We'll need some tests showing it working.

Do you mean we need UT or IT, or just test plan for this?

**Comment by bitwiseman** [comment]
@gmshake 
I mean we need tests that pass values and exercise this new code.  Unit tests in this project at minimum.

**Comment by gmshake** [comment]
Hi, @bitwiseman 

1. Rebased to lastest master.
2. Updated the since tag / comments

> I mean we need tests that pass values and exercise this new code. Unit tests in this project at minimum.

3. Unit tests added.

**Comment by gmshake** [comment]
Looks the commit message need be amended.

**Comment by gmshake** [comment]
ping @bitwiseman

**Comment by jmendiara** [comment]
Really interested in this PR moving fordward. @bitwiseman would you mind to have a look, please?

**Comment by gmshake** [comment]
Close as @agabrys has fixed https://github.com/jenkinsci/branch-api-plugin/pull/186 by https://github.com/jenkinsci/branch-api-plugin/pull/244 .

**Pull request closed without merging**